### DISCUSSION TITLE
Repo Install Loacation

### DISCUSSION THREAD

**Discussion by Ducci9129** [Cog-Creators/Red-DiscordBot#6179] [Q&A] [answered]
When a repo is installed are the contents of the repo moved to the local machine, or is it put in some list that logs repo URLs to be used when installing/loading a cog? I'm poking around where Red was installed, looking for the location of the repos, and I'm unable to find a folder for the repos to be downloaded to or a list of repo URLs to be searched for the cog being installed.

**Comment by Jackenmen** [comment, marked as answer]
Added repositories are cloned into RepoManager's cog data path (which is part of the instance's data path, as returned by `redbot --debuginfo <instance name>`). This shouldn't need to be touched, it is all managed by the Downloader cog.

**Comment by Flame442** [comment]
The repo is actually stored in two places, the entire repo is cloned to `<datapath>\cogs\RepoManager` when `[p]repo add` is run, and individual cogs are copied to the `Install Path` listed in `[p]paths` when `[p]cog install` is run.

As Jack said, you should never need to touch these files, as they are managed by downloader. If you want to modify a 3rd party cog, the best way to do that is to make a fork of the source repo on Github, commit your changes to your fork, and install your fork using downloader as normal. That way, you can easily pull updates from upstream using normal git tools, and you don't run the risk of accidentally overriding your changes with a `[p]cog update`.

If you need any further support, you can always ask in the [Red support server](https://discord.gg/red).

**Reply by Ducci9129** [reply]
Ty, I'm just messing around with it to figure out how it works.
